src/buildercore/core.py

```diff
@@ -111,13 +107,22 @@ def connect_aws_with_stack(stackname, service):
     return connect_aws_with_pname(pname, service)
 
 def find_ec2_instance(stackname):
-    "returns ec2 instance data for a *specific* stackname"
+    "returns list of ec2 instances data for a *specific* stackname"
     # filters: http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstances.html
-    kwargs = {
+    filter_by_cluster = {
+        'filters': {
+            'tag:Cluster':[stackname],
+            'instance-state-name': ['running']}}
+    filter_by_name = {
         'filters': {
             'tag:Name':[stackname],
             'instance-state-name': ['running']}}
-    return connect_aws_with_stack(stackname, 'ec2').get_only_instances(**kwargs)
+    conn = connect_aws_with_stack(stackname, 'ec2')
+    cluster_result = conn.get_only_instances(**filter_by_cluster)
```

Comment:
we could invert the conditions so that the normal case is 1 instance, and it succeed on the first call. The backward compatibility is towards existing EC2 instances, which do not have the `Cluster` tag